File: Programming EF Code First/chapter3/Chapter 3 VB/Data Annotations/Model/Person.vb
Imports System.ComponentModel.DataAnnotations

Namespace Model
  Public Class Person
	Public Sub New()
	  Address = New Address()
	  Info = New PersonalInfo With {.Weight = New Measurement(), .Height = New Measurement()}
	End Sub

	Public Property PersonId() As Integer
	<ConcurrencyCheck>
	Public Property SocialSecurityNumber() As Integer
	Public Property FirstName() As String
	Public Property LastName() As String
	Public Property Address() As Address
	Public Property Info() As PersonalInfo
  End Class
End Namespace

File: Programming EF Code First/chapter3/Chapter3/BAGA_DataAnnotations/Model/Person.cs
using System.ComponentModel.DataAnnotations;

namespace Model
{
  public class Person
  {
    public Person()
    {
      Address = new Address();
      Info = new PersonalInfo
      {
        Weight = new Measurement(),
        Height = new Measurement()
      };
    }

    public int PersonId { get; set; }
    [ConcurrencyCheck]
    public int SocialSecurityNumber { get; set; }
    public string FirstName { get; set; }
    public string LastName { get; set; }
    public Address Address { get; set; }
    public PersonalInfo Info { get; set; }
  }
}
